gobblin-data-management/src/main/java/gobblin/data/management/copy/PreserveAttributes.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy;
+
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+
+import com.google.common.base.Strings;
+
+
+/**
+* Configuration for preserving attributes in Gobblin distcp jobs.
+*/
+@AllArgsConstructor
+@EqualsAndHashCode
+public class PreserveAttributes {
+
+  /**
+   * Attributes that can be preserved.
+   */
+  public static enum Option {
+    REPLICATION('r'),
+    BLOCK_SIZE('b'),
+    OWNER('u'),
+    GROUP('g'),
+    PERMISSION('p');
+
+    private final char token;
+
+    Option(char token) {
+      this.token = token;
+    }
+  }
+
+  private int options;
+
+  /**
+   * @return true if attribute should be preserved.
+   */
+  public boolean preserve(Option option) {
+    return 0 < (this.options & (1 << option.ordinal()));
+  }
+
+  /**
+   * Parse {@link PreserveAttributes} from a string of the form [rbugp]:
+   * * r -> preserve replication
+   * * b -> preserve block size
+   * * u -> preserve owner
+   * * g -> preserve group
+   * * p -> preserve permissions
+   *
+   * @param s String of the form [rbugp]
```

Comment:
Are the brackets part of the string? Should we also have the shortcut "a" for all?